src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -127,11 +131,16 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             // Ensure the options persisters are loaded since we have to fetch options from the shell
             _componentModel.GetExtensions<IOptionPersister>();
 
-            _workspace = _componentModel.GetService<VisualStudioWorkspace>();
+            _workspace = (VisualStudioWorkspaceImpl)_componentModel.GetService<VisualStudioWorkspace>();
             _workspace.Services.GetService<IExperimentationService>();
 
             RoslynTelemetrySetup.Initialize(this, telemetrySession);
 
+            var provider = new VisualStudioDiagnosticAnalyzerProvider(this);
+
+            // fire and forget
+            _ = Task.Run(() => InitializeHostAnalyzerReferences(provider));
```

Comment:
Was this previously async? If it wasn't this may risk some flakiness in integration tests if we have any. Surround it with an async listener operation so tests can appropriately wait?

Reply:
Added. I assume integration tests need to be updated to wait for it. Where should such wait be done?